### ISSUE TITLE
Accessibility Issues - Dropdown component - Carbon IOT React level issue

### ISSUE THREAD

**Issue by agupta2-ibm-maximo** [IBM/carbon-addons-iot-react#616]
Getting 2 accessibility issues for Dropdown component in Graphite. Issues are due to Carbon react level.

Issues are:
1. **Ensures every ARIA input field has an accessible name**

ARIA input fields have an accessible name

More info...
#dropdown

Serious
aria-label attribute does not exist or is empty
Serious
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
Serious
Element has no title attribute or the title attribute is empty
wcag2a
wcag412

2. **Ensures elements with an ARIA role that require child roles contain them**

Certain ARIA roles must contain particular children

More info...
#dropdown

Critical
Required ARIA child role not present: option
cat.aria
wcag2a
wcag131

Observation:
As per carbon design react storybook "http://react.carbondesignsystem.com/?path=/story/dropdown--default" if we remove the content  from the knob "Title (titleText)", we will start getting same error in DAP.
 
Details:

As per carbon design react storybook below elements got rendered when text is avaialble in knob "Title (titleText)"-
 
```
<label for="carbon-dropdown-example" class="bx--label" data-namewalk="30">This is not a dropdown title.</label>
<div id="carbon-dropdown-example" role="listbox" tabindex="-1" class="bx--dropdown bx--list-box" data-namewalk="32">
```
 
*label element is defining label for dropdown id "carbon-dropdown-example".
 
After clearing out the content of knob "Title (titleText)", we get below html -
 
```
<div id="carbon-dropdown-example" role="listbox" tabindex="-1" class="bx--dropdown bx--list-box" data-namewalk="32">
```
 
*no label element to define label for dropdown id "carbon-dropdown-example".
 
If we try to run DAP after clearing out the knob content as stated above, we get same accessibility violation.
 
Summary:
carbon design react component expect us to have title whenever we use a dropdown but as per our application title is not mandatory to be with every dropdown used in the application.

(url
[Dropdown_Component_Issue.docx](https://github.com/IBM/carbon-addons-iot-react/files/3744067/Dropdown_Component_Issue.docx)
)

**Comment by davidicus** [comment]
Hi @agupta2-ibm-maximo. Thanks for submitting this issue.

**Comment by enricoberti** [comment]
hi @agupta2-ibm-maximo,
the first problem has been fixed already on Carbon while I opened https://github.com/carbon-design-system/carbon/issues/5368 to fix the second

**Comment by enricoberti** [comment]
A much bigger change from the Carbon team is on its way https://github.com/carbon-design-system/carbon/pull/5373 
That solves our problem plus many more a11y related
